File: zad2/app/src/main/java/com/example/zad_2/ui/page1/FirstViewModel.java
package com.example.zad_2.ui.page1;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

public class FirstViewModel extends ViewModel {
    private final MutableLiveData<String> mText;
    public FirstViewModel() {
        mText = new MutableLiveData<>();
        mText.setValue("This is first page");
    }
    public LiveData<String> getText() {
        return mText;
    }
}

File: zad4/app/src/main/java/com/example/zad4/ui/page1/FirstViewModel.kt
package com.example.zad4.ui.page1

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel

class FirstViewModel : ViewModel() {
    private val mText: MutableLiveData<String>

    init {
        mText = MutableLiveData()
        mText.value = "This is first page"
    }

    val text: LiveData<String>
        get() = mText
}